sync/index.test.ts

```diff
@@ -226,6 +226,41 @@ test('maps input actions', async () => {
   equal(pair.rightNode.log.actions(), [{ type: 'a1' }])
 })
 
+test('handles error in onSync', async () => {
+  let error = new Error('test')
+  let catched: Error[] = []
+
+  let pair = await createTest()
+  pair.rightNode.options.onSync = () => {
+    throw error
+  }
+  pair.rightNode.catch(e => {
+    catched.push(e)
+  })
+  pair.leftNode.log.add({ type: 'a' })
+
+  await delay(50)
+  equal(catched, [error])
+})
+
+test('onSync is called instead of adding an action to the log', async () => {
+  let actions = []
+  let pair = await createTest(created => {
+    created.rightNode.options.onSync = (action, meta) => {
```

Comment:
I mean let’s remove `type(action, 'object')` from `inMap`/`inFilter`.

The `inMap`/`inFilter` were useful to test that we call `onSync` after them.